go/tdh2/tdh2easy/tdh2easy.go

```diff
@@ -194,22 +194,83 @@ type ciphertextRaw struct {
 	Nonce    []byte
 }
 
+// ciphertextRaw is serialized as _TDH2Ctxt || _SymCtxt || _Nonce
+// where _TDH2Ctxt, _SymCtxt, and _Nonce are length-prefixed byte slices.
+
+func (c ciphertextRaw) Marshal() ([]byte, error) {
+	buf := make([]byte, 0, 4+len(c.TDH2Ctxt)+4+len(c.SymCtxt)+4+len(c.Nonce))
```

Comment:
len() is 64bit on 64bit architectures, so assuming 4 bytes for length is incorrect